src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs

```diff
@@ -87,8 +83,8 @@ private SafeDirectoryHandle CreateDirectoryHandle(string path)
 
         private void CloseDirectoryHandle()
         {
-            _directoryHandle?.Dispose();
-            _directoryHandle = null;
+            Interop.Sys.CloseDir(_directoryHandle);
+            _directoryHandle = IntPtr.Zero;
         }
```

Comment:
There's a finalizer somewhere that ensures this is called?

Reply:
Yes, in the shared file.

Reply:
Yes, on FileSystemEnumerator